build/common/installer/scripts/tomlparser-common-agent-config.rb

```diff
@@ -62,12 +63,25 @@ def populateSettingValuesFromConfigMap(parsedConfig)
   end
 end
 
+def getKubernetesMetadataCacheTTL(parsedConfig)
+  begin
+    if !parsedConfig.nil? && !parsedConfig[:agent_settings].nil?
+      k8s_metadata_config = parsedConfig[:agent_settings][:k8s_metadata_config]
+      if !k8s_metadata_config.nil? && !k8s_metadata_config[:kube_meta_cache_ttl].nil?
+        @logEnableKubernetesMetadataCacheTTL = k8s_metadata_config[:kube_meta_cache_ttl]
```

Comment:
check the provided  value in the config is integer or not  and also check for the value is >=0

Reply:
changed